From 04be12224f948ab5460e966c053100877a271843 Mon Sep 17 00:00:00 2001
From: John Greene <jogreene@redhat.com>
Date: Thu, 14 Aug 2014 14:42:20 -0400
Subject: [ethernet] e1000e: Fix expand setting EEE link info to all affected parts

Message-id: <1408027348-9610-27-git-send-email-jogreene@redhat.com>
Patchwork-id: 87730
O-Subject: [RHEL7.1 PATCH 26/34] e1000e: Fix expand setting EEE link info to all affected parts
Bugzilla: 1091119
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Previously, the update_phy_task was only calling e1000_set_eee_pchlan()
for phy.type 82579.  This patch is to cause this function to be called
for 82579 and newer phy.types.  This causes the dev_spec->eee_lp_ability
to have the correct value when going into SX states.

Signed-off-by: Dave Ertman <davidx.m.ertman@intel.com>
Tested-by: Aaron Brown <aaron.f.brown@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
(cherry picked from commit 50844bb7f445165afda1af315bd16c93bf528185)

BZ: 1091119
Testing: Acceptable, see BZ
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7834849
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/e1000e/netdev.c b/drivers/net/ethernet/intel/e1000e/netdev.c
index b438130..d2f731c 100644
--- a/drivers/net/ethernet/intel/e1000e/netdev.c
+++ b/drivers/net/ethernet/intel/e1000e/netdev.c
@@ -4520,7 +4520,7 @@ static void e1000e_update_phy_task(struct work_struct *work)
  e1000_get_phy_info(hw);
 
  /* Enable EEE on 82579 after link up */
- if (hw->phy.type == e1000_phy_82579)
+ if (hw->phy.type >= e1000_phy_82579)
   e1000_set_eee_pchlan(hw);
 }
 
-- 
1.7.1